packages/menu/src/MenuItem.ts

```diff
@@ -570,38 +592,37 @@ export class MenuItem extends LikeAnchor(
     }
 
     protected async handleSubmenuPointerleave(): Promise<void> {
-        requestAnimationFrame(() => {
-            this.recentlyLeftChild = false;
-        });
+        this.recentlyLeftChild = false;
     }
 
     protected handleSubmenuOpen(event: Event): void {
-        const shouldFocus = this.matches(':focus, :focus-within') || this.focused;
-        this.focused = false;
         const parentOverlay = event.composedPath().find((el) => {
             return (
                 el !== this.overlayElement &&
                 (el as HTMLElement).localName === 'sp-overlay'
             );
         }) as Overlay;
-        if (shouldFocus)
+        if (this._openedViaKeyboard) {
             this.submenuElement?.focus();
+        }
         this.overlayElement.parentOverlayToForceClose = parentOverlay;
     }
 
     protected cleanup(): void {
+        this._closedViaPointer = false;
         this.setAttribute('aria-expanded', 'false');
         this.open = false;
         this.active = false;
     }
 
-    public async openOverlay(): Promise<void> {
+    public async openOverlay(shouldFocus: boolean = false): Promise<void> {
```

Comment:
nice addition! I love the clarity here of opting into focus on overlay open instead of defaulting it.